### PR TITLE
Code review of entire module for transfer to Rally

### PR DESCRIPTION
**Pull request by jeffmay**
@rallyhealth/Engineers @arkban @usufruct99 @slustbader @galak-fyyar @rh-nalitkin

This is every commit from the beginning of time in a PR, so that comments can be made. I will address any comments in a separate PR.

Please review the code to make sure it meets Rally Health's code quality standards.

Note: The license is Apache 2.0, but it might need to change to MIT. This is not visible in this PR, because the [LICENSE](https://github.com/jeffmay/play-json-ops/blob/master/LICENSE) file was added in the first commit to the repo.

Depends on changes:
- [x] https://github.com/rallyhealth/play-json-ops/pull/22 - Change of license and published artifact location

### CODE REVIEW

**Comment by arkban** [comment]
Um, this is on the "jeffmay" org, shouldn't it be on the "Rallyhealth" org?

**Comment by arkban** [comment]
> Note: The license is Apache 2.0, but it might need to change to MIT. 

I thought APL was okay.

File: .gitignore [hunk 1]
```diff
@@ -12,6 +12,9 @@ src_managed/
 project/boot/
 project/plugins/project/
 
+# IntelliJ Specific
```
Comment (arkban) [inline .gitignore:15]: I've heard good arguments that this should be in users global `.gitignore`

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
+language: scala
```
Comment (arkban) [inline .travis.yml:1]: If this is going to Rally I think we're going to want to use CJE instead of travis or anything else.

File: LICENSE [hunk 3]
```diff
@@ -186,7 +186,7 @@ Apache License
       same "printed page" as the copyright notice for easier
       identification within third-party archives.
 
-   Copyright {yyyy} {name of copyright owner}
+   Copyright 2015 Jeff May
```
Comment (arkban) [inline LICENSE]: Um...

File: README.md [hunk 4]
```diff
@@ -1,2 +1,372 @@
...
+- Formats for all tuples as JsArray
+- ScalaCheck generators for JsValue, JsArray, and JsObject
+- Formats for scala.concurrent.Duration
+- UTCFormats for org.joda.time.DateTime
+- Compile-time Json.oformat and Json.owrites macros (Play 2.3 only)
+
+<h1>Versions</h1>
+
+<h2>2.X Branch</h2>
+<table>
```
Comment (arkban) [inline README.md:21]: You know you can write tables in markdown, right?

File: build.sbt [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+      "-Xfatal-warnings",
+      "-deprecation:false",
+      "-feature",
+      "-Ywarn-dead-code",
+      "-encoding", "UTF-8"
+    ),
+
+    resolvers ++= Seq(
+      "Typesafe Releases" at "http://repo.typesafe.com/typesafe/releases/",
+      "jeffmay at bintray" at "https://dl.bintray.com/jeffmay/maven"
```
Comment (arkban) [inline build.sbt]: Um...

File: build.sbt [hunk 6]
```diff
@@ -0,0 +1,96 @@
+name := "play-json-ops-root"
+organization in ThisBuild := "me.jeffmay"
```
Comment (arkban) [inline build.sbt]: I'm not sure the details of the transfer, but if you are giving ownership to Rally this should probably be Rally

File: project/Dependencies.scala [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+  def playJson(playVersion: String): ModuleID = {
+    "com.typesafe.play" %% "play-json" % playVersion
+  }
+
+  def scalacheckOps(scalatestVersion: String): ModuleID = {
+    val suffix = scalatestVersion match {
+      case `scalatest2Version` => ""
+      case `scalatest3Version` => "_1-13"
+    }
+    "me.jeffmay" %% s"scalacheck-ops$suffix" % scalacheckOpsVersion
```
Comment (arkban) [inline project/Dependencies.scala]: Is this correct?

**Review by arkban — CHANGES_REQUESTED** [review]
I reviewed the files in the root directory, but that's it, not the code. If I get time I can try to do that later.

File: LICENSE (hunk 3 above)
Comment (jeffmay) [reply LICENSE]: I can update this before transferring ownership of the repo.

File: .travis.yml (hunk 2 above)
Comment (jeffmay) [reply .travis.yml:1]: We are currently using travis for all of our open-source Scala projects. We can discuss changing this in a separate thread.

File: .gitignore (hunk 1 above)
Comment (jeffmay) [reply .gitignore:15]: Probably... but I can't imagine anyone not using IntelliJ for Scala 😲

File: README.md (hunk 4 above)
Comment (jeffmay) [reply README.md:21]: I remember when I first attempted to do it in markdown, it wasn't powerful enough to do what I wanted, so I used HTML. I can give it another try. Maybe a fast-follow item?

File: build.sbt (hunk 5 above)
Comment (jeffmay) [reply build.sbt]: This review is about transferring ownership to Rally. It isn't published to Rally currently. In fact, this PR isn't about merging anything. It's just about reviewing the code. I can see the confusion though. How about I transfer the repo to Rally Health and then do a review?

File: .gitignore (hunk 1 above)
Comment (arkban) [reply .gitignore:15]: Sounds like a challenge for Vito ;)

File: .travis.yml (hunk 2 above)
Comment (arkban) [reply .travis.yml:1]: TIL, no need to discuss if it's the way we do things for our OSS projects

File: build.sbt (hunk 5 above)
Comment (arkban) [reply build.sbt]: Either that or just clarify, because I (clearly) didn't understand that.

File: README.md (hunk 4 above)
Comment (arkban) [reply README.md:21]: It's fine, just letting you know, doesn't need to be a fast follow, sounds more like an enhancement that will come when someone gets fed up modifying the HTML table ;p

**Pull request closed without merging**